src/SprykerEco/Yves/Heidelpay/Theme/default/components/molecules/direct-debit-new-registration/direct-debit-new-registration.ts

```diff
@@ -0,0 +1,73 @@
+import Component from 'ShopUi/models/component';
+
+export default class DirectDebitNewRegistration extends Component {
+
+    protected inputs: HTMLInputElement[];
+    protected paymentForm: HTMLFormElement;
+    protected paymentMethodToggler: HTMLInputElement;
+    protected paymentFormTogglers: HTMLInputElement[];
+    protected directDebitPaymentToggler: HTMLInputElement;
+    protected paymentName: string = 'heidelpayDirectDebit';
+    protected paymentOptionsTogglers: HTMLInputElement[];
+
+    protected readyCallback(): void {}
+
+    protected init(): void {
+        this.paymentForm = <HTMLFormElement>document.getElementById(this.parentFormId);
+        this.inputs = Array.from(this.getElementsByTagName('input'));
+        this.paymentOptionsTogglers = <HTMLInputElement[]>Array.from(this.paymentForm.querySelectorAll(`input[name="${this.paymentOptionInputName}"`));
+        this.directDebitPaymentToggler = <HTMLInputElement>this.paymentOptionsTogglers.find(toggler => toggler.value === this.name);
+        this.paymentFormTogglers = <HTMLInputElement[]>Array.from(this.paymentForm.querySelectorAll(this.paymentTogglerSelector));
+        this.paymentMethodToggler = <HTMLInputElement>this.paymentFormTogglers.find(toggler => toggler.value === this.paymentName);
+        this.mapEvents();
+    }
+
+    protected mapEvents(): void {
+        this.paymentForm.addEventListener('submit', this.handleSubmit.bind(this));
+    }
+
+    protected handleSubmit(event: Event): void {
+        if (this.isDirectDebitNewRegistrationActive()) {
+            event.preventDefault();
+            this.setupFormAttributes();
+            this.filterFormFields();
+            this.paymentForm.submit();
+        }
+    }
+
+    protected filterFormFields(): void {
+        const elements = Array.from(this.paymentForm.elements) as HTMLInputElement[];
+        elements.forEach(element => {
+            this.inputs.forEach(input => {
+                element.disabled = true;
+                if (element.name === input.name || element.type === 'submit') {
+                    element.disabled = false;
+                }
+            });
+        });
+    }
+
+    protected setupFormAttributes(): void {
```

Comment:
You are not setting up form attributes, you are changing the form action.

Reply:
fixed